### PR TITLE
fix(joins): check reachability before sending msgs

### PR DESCRIPTION
**Pull request by oetyng**
We tried to send a msg to the peer that failed the check. That wouldn't work though.
Also, other branches in same flow were sending msgs before doing that check.

### CODE REVIEW

**Comment by joshuef** [comment]
> We tried to send a msg to the peer that failed the check. That wouldn't work though.

I think we'd be able to send it back on the initial connection. (The comm check attempts to open a _new_ conn I think, to ascertain reachability)

File: sn_node/src/node/messaging/system_msgs.rs [hunk 1]
```diff
@@ -400,7 +407,13 @@ impl Node {
                     return Ok(vec![]);
                 }
 
-                self.handle_join_as_relocated_request(sender, *join_request, comm)
+                // First do reachability check (we can't respond to the node if it fails).
+                if comm.is_reachable(&sender.addr()).await.is_err() {
+                    debug!("Ignoring JoinAsRelocatedRequest from {sender} - not reachable.");
```
Comment (Yoga07) [inline sn_node/src/node/messaging/system_msgs.rs:412]: Would nodes on relocation stall if they don't get a response back? Think we should add a timeout there alongside this change maybe?

It won't be the same case for normal joins as they have a timeout and exit the process if there's no response, but the flow is a bit different here for relocations.

Comment (Yoga07) [reply sn_node/src/node/messaging/system_msgs.rs:412]: Even without this change, I think we should have a timeout during relocations(or at least a warning of the time it takes) as they are the same as regular joins.

Comment (oetyng) [reply sn_node/src/node/messaging/system_msgs.rs:412]: > I think we'd be able to send it back on the initial connection. (The comm check attempts to open a _new_ conn I think, to ascertain reachability)

Mm, that's right, the incoming conn is (most likely) still there, and there msgs _can_ be received. Well, then there's no need to move the check.

**Comment by oetyng** [comment]
> I think we'd be able to send it back on the initial connection. (The comm check attempts to open a new conn I think, to ascertain reachability)

Mm, that's right, the incoming conn is (most likely) still there, and there msgs can be received. Well, then there's no need to move the check.

**Comment by oetyng** [comment]
The incoming conn is (most likely) still there, and there msgs can be received.

**Pull request closed without merging**